bot/exts/previewing/preview.py

```diff
@@ -37,14 +37,16 @@ async def server_icon(
         icon = (await download_image(file_url)).resize(ICON_SIZE)
         icon = add_background(icon, Preview.background_color(mode.lower()))  # type: ignore
 
-        with Image.open(ICON_TEMPLATES.format(mode=mode)) as template:
-            preview = Image.new("RGBA", template.size)
-            for position in ICON_POSITIONS:
-                preview.paste(icon, position)
+        def long_stuff():
+            with Image.open(ICON_TEMPLATES.format(mode=mode)) as template:
+                preview = Image.new("RGBA", template.size)
+                for position in ICON_POSITIONS:
+                    preview.paste(icon, position)
 
-            with Image.open(ICON_TEMPLATES.format(mode="Mask")) as mask:
-                mask = mask.convert("L")
-                preview = Image.composite(preview, template, mask)
+                with Image.open(ICON_TEMPLATES.format(mode="Mask")) as mask:
+                    mask = mask.convert("L")
+                    return Image.composite(preview, template, mask)
+        preview = await in_executor(long_stuff)
 
         await inter.response.send_message(file=image_to_file(preview))
```

Comment:
```suggestion

        preview = await in_executor(long_stuff)
        await inter.response.send_message(file=image_to_file(preview))
```
I think this separation makes more sense.